hyperspy/component.py

```diff
@@ -1224,3 +1226,67 @@ def print_current_values(self, only_free=False, fancy=True):
         else:
             display_pretty(current_component_values(
                 self, only_free=only_free))
+
+    @property
+    def is_linear(self):
+        """Loops through the components free parameters,
+        checks that they are linear"""
+        linear = True
+        for para in self.free_parameters:
+            if not para._is_linear:
+                linear = False
+        return linear
+
+    @property
+    def linear_parameters(self):
+        return [para for para in self.free_parameters if para._is_linear]
+
+    @property
+    def nonlinear_parameters(self):
+        return [para for para in self.free_parameters if not para._is_linear]
+
+    @property
+    def _constant_term(self):
+        """Get value of any (non-free) constant term of the component.
+        Returns 0 for most components."""
+        return 0
+
+    def _compute_component(self):
+        model = self.model
+        if model.convolved and self.convolved:
+            # TODO: Model2D doesn't support a 2D convolution axis (yet)
```

Comment:
Why the TODO?

Reply:
This seems like it has changed, since now `hs.signals.Signal2D([[1]]).create_model().convolved` doesn't raise an error. I originally kept it there as a reminder (when I should have written an issue). I'll look into it and remove it/write an issue about it.